p2p/switch.go

```diff
@@ -409,20 +408,22 @@ func (s *Switch) sendMessageImpl(peerPubKey p2pcrypto.PublicKey, protocol string
 
 // RegisterDirectProtocol registers an handler for a direct messaging based protocol.
 func (s *Switch) RegisterDirectProtocol(protocol string) chan service.DirectMessage { // TODO: not used - remove
+	if s.started == 1 {
+		panic("Calling register after the p2p has started")
```

Comment:
I would use `log.Panic` here for consistency across the project and for better support for logging panics.